python/mspasspy/db/database.py

```diff
@@ -1089,3 +1101,125 @@ def load_event(self, source_id):
+       or bson.objectid.ObjectId.
+
+    :return: dict containing the repaired (key,value) pair.  If repair was not possible key
+    will be set to 'FAILURE' and the value will be a string with an
+    informational message the user may choose to print or post to an
+    error log.
+
+    :exception:  Can throw an exception ONLY if the required_type field is
+      not in the list of supported types for MsPASS.  It is a coding error
+      created in maintenance if this throws an exception.
+
+    """
+    try:
+        if required_type is str:
+            x=str(value)
+        elif required_type is int:
+            x=int(value)
+        elif required_type is float:
+            x=float(value)
+        elif required_type is bool:
+            # Note I don't think bool will ever throw a type error but
+            # is subject to confusing behavior if user doesn't realize
+            # the are dealing with a bool
+            x=bool(value)
+        elif required_type is ObjectId:
+            x=ObjectId(value)
+        elif required_type is list:
+            #This needs to throw an exception because it is nearly
+            # guaranteed to be an error caused by setting something incorrectly.
+            raise TypeError("Entry with key="+key
+                +" is expected to be a python list but is of type"+type(value)
+                +"\nSomething probably improperly set this reserved key\n"
+                +"Need to abort or this will cause downstream problems if used")
+        else:
+            raise TypeError("Error in schema - defines unsupported type="
+                            +str(required_type))
+        return {key : x}
+    except ValueError as err:
+        error_message="Failure attempting to convert value with key="+key
+        error_message+="\nValueError message python posted:  "+str(err)
+        return {'FAILURE' : error_message}
+    #except bson.objectid.InvalidId as bsonerr:
+    except TypeError as bsonerr:
+        error_message="Failure attempting to convert value with key="+key
+        error_message+=" to an ObjectId as the schema requires\n" + "Message posted in python:  "+str(bsonerr)
+        return {'FAILURE' : error_message}
+def _sync_time_metadata(mspass_object):
+    """
+    MsPASS data objects are designed to cleanly handle what we call relative
+    and UTC time.  This small helper function assures the Metadata of
+    mspass_object are consistent with the internal contents.  That
+    involves posting some special attributes seen below to handle this issue.
+    Since Metadata is volatile we need to be sure these are consistent or
+    timing can be destroyed on data.
+    """
+    # this adds a small overhead but it guarantees Metadata and internal t0
+    # values are consistent.  Shouldn't happen unless the user messes with them
+    # incorrectly, but this safety is prudent to reduce the odds of mysterious
+    # timing errors in data
+    t0=mspass_object.t0
+    mspass_object.set_t0(t0)
```

Comment:
Note here, this is actually what revealed the bug with `t0`

Reply:
@pavlis Actually, I am confused. Here we save the `utc_convertible` and `startime_shift` into the database, but how are we gonna use them when reading data? These will be retrieved to the Metadata, but the `ator` and `rtoa` methods won't use the Metadata field. Maybe we should make the `t0shift_is_valid` and `t0shift` of `BasicTimeSeries` visible and link them to the corresponding Metadata field?

Reply:
We should continue this topic in #78